satpy/resample/__init__.py

```diff
@@ -0,0 +1,209 @@
+
+    >>> from pyresample.geometry import AreaDefinition
+    >>> my_area = AreaDefinition(...)
+    >>> local_scene = scn.resample(my_area)
+
+Resize area definition in pixels
+--------------------------------
+
+Sometimes you may want to create a small image with fixed size in pixels.
+For example, to create an image of (y, x) pixels :
+
+    >>> small_scn = scn.resample(scn.finest_area().copy(height=y, width=x), resampler="nearest")
+
+
+.. warning::
+
+    Be aware that resizing with native resampling (``resampler="native"``) only
+    works if the new size is an integer factor of the original input size. For example,
+    multiplying the size by 2 or dividing the size by 2. Multiplying by 1.5 would
+    not be allowed.
+
+
+Create dynamic area definition
+------------------------------
+
+See :class:`pyresample.geometry.DynamicAreaDefinition` for more information.
+
+Examples coming soon...
+
+Store area definitions
+----------------------
+
+Area definitions can be saved to a custom YAML file (see
+`pyresample's writing to disk <http://pyresample.readthedocs.io/en/stable/geometry_utils.html#writing-to-disk>`_)
+and loaded using pyresample's utility methods
+(`pyresample's loading from disk <http://pyresample.readthedocs.io/en/stable/geometry_utils.html#loading-from-disk>`_)::
+
+    >>> from pyresample import load_area
+    >>> my_area = load_area('my_areas.yaml', 'my_area')
+
+Or using :func:`satpy.resample.base.get_area_def`, which will search through all
+``areas.yaml`` files in your ``SATPY_CONFIG_PATH``::
+
+    >>> from satpy.base.resample import get_area_def
+    >>> area_eurol = get_area_def("eurol")
+
+For examples of area definitions, see the file ``etc/areas.yaml`` that is
+included with Satpy and where all the area definitions shipped with Satpy are
+defined. The section below gives an overview of these area definitions.
+
+Area definitions included in Satpy
+----------------------------------
+
+.. include:: /area_def_list.rst
+
+"""
+from __future__ import annotations
+
+import warnings
+from typing import Any
+
+
+def __getattr__(name: str) -> Any:
+    if name in ("KDTreeResampler", "BilinearResampler"):
+        from . import kdtree
+
+        new_submod = "kdtree"
+        obj = getattr(kdtree, name)
+    elif name == "NativeResampler":
+        from .native import NativeResampler
+        new_submod = "native"
+        obj = NativeResampler
+    elif name in (
+            "BucketResamplerBase",
+            "BucketAvg",
+            "BucketSum",
+            "BucketCount",
+            "BucketFraction"
+    ):
+        from . import bucket
+        new_submod = "bucket"
+        obj = getattr(bucket, name)
+    elif name in (
+            "get_area_file",
+            "get_area_def",
+            "add_xy_coords",
+            "add_crs_xy_coords",
+            "update_resampled_coords",
+            "resample",
+            "prepare_resampler",
+            "get_fill_value",
+            "resample_dataset"
```

Comment:
Are all of these used outside of the resample code? I know Scene calls `prepare_resampler` and `resample_dataset`, but are the other functions used? Maybe they could be made private?

Reply:
I made `update_resampled_coords` and `get_fill_value` private. The former is used only in AWIPS tests outside of `resample` package, the latter only within resamplers. The others have more uses also elsewhere.